lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -8,641 +8,395 @@ module Leeds
+              @teststatus = 9
+            elsif @report.include?('The variant c.1409+47T>C was detected in intron 12')
+              @teststatus = 10
+            elsif @report.include?('sequence variant c.1595G>A')
+              @teststatus = 2
+            end
+          end
+
+          def cleanup_report(report)
+            raw_report = report
+            exclude_statements = [
+              'Screening for mutations in MLH1, MSH2 and MSH6 is now in progress as requested.',
+              'MLPA and MSH2 analysis was not requested.',
+              'MLPA and MSH2 analysis were not requested.',
+              'if MSH2 and MSH6 data analysis is required.',
+              'No further screening for mutations in MLH1, MSH2 or MSH6 has been performed.',
+              'developing further MSH2-related cancers',
+              'developing MSH2-associated cancer'
+            ]
+
+            exclude_statements.each { |st| raw_report&.sub!(st, '') }
+            raw_report
+          end
+
+          def get_genes_from_report(report)
+            genes = report&.scan(COLORECTAL_GENES_REGEX)&.flatten&.uniq
+            if genes.present?
+              genes -= ['Met'] if @report.scan(/p\.\(?\w*Met/).size.positive?
+              genes -= ['met'] if @report.scan('endometrial').size.positive?
+            end
+            genes || []
+          end
+
+          def add_organisationcode_testresult(genocolorectal)
+            genocolorectal.attribute_map['organisationcode_testresult'] = '699C0'
+          end
+
+          def add_varclass
+            @var_class = nil
+            class5 = ->(id) { VARIANT_CLASS_5.include? id }
+            class7 = ->(id) { VARIANT_CLASS_7.include? id }
+            @var_class = case @geno
+                         when class7
+                           var_class_from_report
+                         when /class\s?2/i, 'fap diagn poss benign', 'fap p.glu1317gln'
+                           2
+                         when /class\s?3|c\s?3/i, 'conf mlpa +ve (apc)', 'fap diagn unknown'
+                           3
+                         when /class\s?4|c\s?4/i, 'fap diagn poss path', 'two mutations'
+                           4
+                         when /class\s?5|c\s?5/i, class5
+                           5
+                         end
+          end
+
+          def var_class_from_report
+            report_cleaned = @report.gsub('No further pathogenic changes were detected', '')
+            if report_cleaned.match(PATHOGENIC_REGEX)
+              $LAST_MATCH_INFO[:pathogenic].strip
+            else
+              7
+            end
+          end
+
+          def add_scope(genocolorectal, record)
+            genotype = record.raw_fields['genotype'].downcase.strip
+            mtype = record.raw_fields['moleculartestingtype'].downcase.strip
+            if genotype.match(/pms2\s-\smlpa\spred\snegative\sc5/i) ||
+               genotype.match(/pms2\s-\sconf\smlpa\spositive\sc5/i)
```

Comment:
I can't see this in the rules? Can you send me the updated ones please?

Reply:
Have mailed you the rules , and following - 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/shilpigoel/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/shilpigoel/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{color:#7030A0;
	text-align:left;
	vertical-align:middle;
	border:.5pt solid windowtext;
	background:#D9D9D9;
	mso-pattern:black none;}
.xl66
	{color:#7030A0;
	font-weight:700;
	text-align:left;
	vertical-align:middle;
	border:.5pt solid windowtext;}
.xl67
	{color:#7030A0;
	vertical-align:middle;
	border:.5pt solid windowtext;}
.xl68
	{color:#7030A0;
	font-weight:700;
	vertical-align:middle;
	border:.5pt solid windowtext;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">


Raw:genotype | Test Scope
-- | --
PMS2 - MLPA   pred negative C5 | Targeted
PMS2 - Conf   MLPA positive C5 | Targeted



</body>

</html>

Reply:
Thank you! This makes sense now